src/intents/DSL/schema.ts

```diff
@@ -55,12 +60,39 @@ export const nonceConditionSchema = joi.object({
 
 const priceOperators = Object.values(PriceOperator)
 
+const toTokenWithDecimals = (value: any, helpers: any) => {
+  const chainId = helpers.prefs.context.chainId;
```

Comment:
is it possible this might not be set in context? maybe throw an error here if so

Reply:
done